src/pages/index.js

```diff
@@ -33,53 +48,120 @@ const cardSection = new Section({
   }
 }, ".cards");
 const popupWithImage = new PopupWithImage('.modal_type_image');
-const api = new Api({
-  baseUrl: "https://around-api.en.tripleten-services.com/v1",
-  headers: {
-    authorization: "32400e4e-bdb6-4554-9f2b-339b0c55f4e6",
-    "Content-Type": "application/json"
-  }
-});
+
 
 
 // define functions
 function renderProfileDetails(){
-  const { name, description } = userInfo.getUserInfo();
+  const { name, about } = userInfo.getUserInfo();
   nameInput.value = name;
-  jobInput.value = description;
+  jobInput.value = about;
 };
 
 function handleCardAdd(data){
-  const { title, url } = data
-  const cardElement = createCard({title, url})
-  cardSection.addItem(cardElement)
-  cardModal.close()
+  const submitButton = cardAddFormElement.querySelector('.modal__save-button')
+  const text = submitButton.textContent;
+  submitButton.textContent = "Saving..."
+
+  const { name, link } = data
+  api.addCard(name, link).then((data)=>{
+    cardSection.addItem(createCard(data))
+  }).catch(err => {
+    console.error(err)
+  }).finally(()=>{
+    submitButton.textContent = text;
+    cardModal.close()
+  });
 };
 
 function handleImageClick(image){
   const data = {
-    name: image.title,
-    link: image.url
+    name: image.name,
+    link: image.link
   }
   popupWithImage.open(data)
 };
 
+function handleDeleteClick(card){
+  deleteConfirmModal.open()
+  
+  deleteConfirmModal.setSubmitHandler(()=>{
+    api.deleteCard(card.getCardId()).then(()=>{
+      card.delete()
+      deleteConfirmModal.close()
+    }).catch(err => {
+      console.error(err)
+    })
+  })
+}
+
 function createCard(cardData){
-  const card = new Card(cardData, '.card-template', handleImageClick);
+  const card = new Card(cardData, '.card-template', handleImageClick, handleDeleteClick, handleLikeClick, handleDislikeClick);
   return card.generateCard()
 };
 
 function handleProfileFormSubmit(data){
-  userInfo.setUserInfo(data);
+  const submitButton = profileFormElement.querySelector('.modal__save-button')
+  const text = submitButton.textContent;
+  submitButton.textContent = "Saving..."
+
+  const {name, description} = data;
+  api.updateProfile(name, description).then(data => {
+    return data
+  }).then(() => {
+    renderUserInfo()
```

Comment:
Please, delete `renderUserInfo()`. You don't need to call it when you update the profile data because the **response** from the server has everything you need